app/src/main/java/fiware/smartcity/marketplace/MarketActivity.java

```diff
@@ -0,0 +1,106 @@
+
+import fiware.smartcity.Application;
+import fiware.smartcity.MainActivity;
+import fiware.smartcity.R;
+
+/**
+ * Created by francisco on 29/03/16.
+ */
+public class MarketActivity {
+    private Context context;
+
+    private static Activity activity;
+
+    private Drawable x, y;
+    private WebView webView;
+
+    public MarketActivity(Context ctx) {
+        context = ctx;
+        activity = Application.mainActivity;
+
+        x = activity.getResources().getDrawable(R.drawable.clear);
+        x.setBounds(0, 0, 50, 50);
+
+        y = activity.getResources().getDrawable(R.drawable.search);
+        y.setBounds(0, 0, 50, 50);
+    }
+
+    public void start() {
+        ViewGroup rootContainer = (ViewGroup) activity.findViewById(R.id.mainFrame);
+        activity.getLayoutInflater().inflate(R.layout.route, rootContainer);
+
+        ViewGroup routeContainer = (ViewGroup) activity.findViewById(R.id.frameRoute);
+        Scene scene1 = Scene.getSceneForLayout(routeContainer, R.layout.activity_market, activity);
+
+        TransitionManager.go(scene1);
+
+        // Create a web view containing the Business API Ecosystem web page
+        webView = (WebView) activity.findViewById(R.id.market);
+
+        webView.setWebViewClient(new WebViewClient());
+        webView.setWebChromeClient(new WebChromeClient() {
+            // Handle window.open requests in order to support PayPal redirection
+            @Override
+            public boolean onCreateWindow(WebView view, boolean isDialog, boolean isUserGesture, Message resultMsg) {
+                webView.removeAllViews();
+                webView.scrollTo(0, 0);
+
+                WebView newView = new WebView(context);
+                newView.getSettings().setJavaScriptEnabled(true);
+
+                newView.setWebViewClient(new WebViewClient());
+                newView.setWebChromeClient(new WebChromeClient() {
+                    @Override
+                    public void onCloseWindow(WebView view) {
+                        // Remove the new view when the window is closed in order to return to the marketplace
+                        ((WebView) view.getParent()).removeView(view);
+                    }
+                });
+
+                // Create dynamically a new view
+                newView.setLayoutParams(new RelativeLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT));
+                webView.addView(newView);
+
+                WebView.WebViewTransport transport = (WebView.WebViewTransport) resultMsg.obj;
+                transport.setWebView(newView);
+                resultMsg.sendToTarget();
+                return true;
+            }
+        });
+
+        webView.getSettings().setJavaScriptEnabled(true);
+        webView.getSettings().setJavaScriptCanOpenWindowsAutomatically(true);
+        webView.getSettings().setSupportMultipleWindows(true);
+
+        webView.loadUrl("https://demo-mwc.conwet.com/");
```

Comment:
I think we need to add some kind of spinner while loading the URL, otherwise we see a blank page and we don't know what is happening